core/src/main/java/tc/oc/pgm/loot/WeightedRandomChooser.java

```diff
@@ -0,0 +1,79 @@
+package tc.oc.pgm.loot;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Random;
+
+public class WeightedRandomChooser<T> {
+
+  private final Map<T, Option> options = new HashMap<>();
```

Comment:
Why not use simply `Map<T, Double>` ? doesn't seem like Option adds much value here except for an extra object allocation